### PR TITLE
Add Harvester v1.2.0 release note

### PR DESCRIPTION
**Pull request by guangbochen**
This is an update followed up on https://github.com/harvester/release-notes/pull/22

**TODO:**
- [x] update doc link from `dev` to `v1.2`

**Preview:**
https://github.com/guangbochen/release-notes/blob/v1.2-gc/v1.2.0.md

### CODE REVIEW

**Comment by LucasSaintarbor** [comment]
@guangbochen Thanks for making these updates! Would you like this one to be the RNs PR that is reviewed / merged and I'll close **https://github.com/harvester/release-notes/pull/22**?

File: v1.2.0.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Harvester v1.2.0 Release Note
+
+This release follows Harvester v1.1.2, the documentation is available at https://docs.harvesterhci.io/v1.2.
```
Comment (LucasSaintarbor) [inline v1.2.0.md:3]:
```suggestion
This release follows Harvester v1.2.0, the documentation is available at https://docs.harvesterhci.io/v1.2.
```

File: v1.2.0.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    - Failed to upgrade 4 nodes from v1.1.2 to v1.2.0, stuck in post-draining on the third node. See [#4285](https://github.com/harvester/harvester/issues/4285).
+
+
+# Dependency Component Versions
+| Component        | Version                                                                         |
+|------------------|---------------------------------------------------------------------------------|
+| Longhorn         | [v1.4.3](https://github.com/longhorn/longhorn/releases/tag/v1.4.3)              |
+| KubeVirt         | [v0.54.0](https://github.com/kubevirt/kubevirt/releases/tag/v0.54.0)            |
+| Embedded Rancher | [v2.7.5](https://github.com/rancher/rancher/releases/tag/v2.6.10)               |
+| RKE2             | [v1.25.9+rke2r1](https://github.com/rancher/rke2/releases/tag/v1.25.9%2Brke2r1) |
```
Comment (LucasSaintarbor) [inline v1.2.0.md:129]:
```suggestion

| Component        | Version                                                                         |
|------------------|---------------------------------------------------------------------------------|
| Longhorn         | [v1.4.3](https://github.com/longhorn/longhorn/releases/tag/v1.4.3)              |
| KubeVirt         | [v0.54.0](https://github.com/kubevirt/kubevirt/releases/tag/v0.54.0)            |
| Embedded Rancher | [v2.7.5](https://github.com/rancher/rancher/releases/tag/v2.6.10)               |
| RKE2             | [v1.25.9+rke2r1](https://github.com/rancher/rke2/releases/tag/v1.25.9%2Brke2r1) |
```

File: v1.2.0.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+@rebeccazzzz
+@starbops
+@TachunLin
+@Vicente-Cheng
+@vickyhella
+@w13915984028
+@WebberHuang1118
+@WuJun2016
+@yaocw2020
+@yasker
```
Comment (LucasSaintarbor) [inline v1.2.0.md:159]:
```suggestion

@albinsun
@bk201
@chrisho
@connorkuehl
@DaiYuzeng
@FrankYang0529
@futuretea
@guangbochen
@ibrokethecloud
@irishgordo
@khushboo-rancher
@lanfon72
@LucasSaintarbor
@mantis-toboggan-md
@markhillgit
@masteryyh
@n313893254
@noahgildersleeve
@rebeccazzzz
@starbops
@TachunLin
@Vicente-Cheng
@vickyhella
@w13915984028
@WebberHuang1118
@WuJun2016
@yaocw2020
@yasker
```

File: v1.2.0.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+- Fail to scale down RKE2 Harvester node driver cluster. See [#4358](https://github.com/harvester/harvester/issues/4358).
+- Failed to pass guest cluster name into Harvester cloud provider. See [#4232](https://github.com/harvester/harvester/issues/4232).
+- Unable to access Node IP addresses over HTTPS on VM router in the Harvester cluster. See [#3960](https://github.com/harvester/harvester/issues/3960) [[Workaround]](https://github.com/harvester/harvester/issues/3960#issuecomment-1560720590).
+- Unable to select Harvester Cloud Provider in Rancher 2.7.x. See [#3750](https://github.com/harvester/harvester/issues/3750) [[Workaround]](https://github.com/harvester/harvester/issues/3750#issuecomment-1498757266).
+- Harvester upgrade may stuck in another operation (install/upgrade/rollback) by fleet. See [#3675](https://github.com/harvester/harvester/issues/3675) [[Workaround]](https://github.com/harvester/harvester/issues/3675#issuecomment-1533211282).
+- Upgrade issues: 
+    - Check https://docs.harvesterhci.io/dev/upgrade/v1-1-2-to-v1-2-0#known-issues for more information.
```
Comment (LucasSaintarbor) [inline v1.2.0.md:111]:
```suggestion
    - Check https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0#known-issues for more information.
```

File: v1.2.0.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    - An upgrade can't start and reports `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`. See [#3839](https://github.com/harvester/harvester/issues/3839) [[Workaround]](https://github.com/harvester/harvester/issues/3839#issuecomment-1534438192).
+    - An upgrade is stuck in `Creating Upgrade Repository`. See [#4246](https://github.com/harvester/harvester/issues/4246) [[Workaround]](https://github.com/harvester/harvester/issues/4246#issuecomment-1695107656)
+    - An upgrade is stuck when pre-draining a node. See [[Workaround]](https://docs.harvesterhci.io/dev/upgrade/v1-1-2-to-v1-2-0#3-an-upgrade-is-stuck-when-pre-draining-a-node).
```
Comment (LucasSaintarbor) [inline v1.2.0.md:114]:
```suggestion
    - An upgrade is stuck when pre-draining a node. See [[Workaround]](https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0#3-an-upgrade-is-stuck-when-pre-draining-a-node).
```

File: v1.2.0.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    - Upgrade is stuck in upgrading the first node: Job was active longer than the specified deadline. See [#2894](https://github.com/harvester/harvester/issues/2894) [[Workaround]](https://github.com/harvester/harvester/issues/2894#issuecomment-1274069690).
+    - An upgrade is stuck when pre-drained a node. See [#3730](https://github.com/harvester/harvester/issues/3730) [[Workaround]](https://docs.harvesterhci.io/dev/upgrade/v1-1-2-to-v1-2-0#5-an-upgrade-is-stuck-in-the-pre-drained-state).
```
Comment (LucasSaintarbor) [inline v1.2.0.md:116]:
```suggestion
    - An upgrade is stuck when pre-drained a node. See [#3730](https://github.com/harvester/harvester/issues/3730) [[Workaround]](https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0#5-an-upgrade-is-stuck-in-the-pre-drained-state).
```

File: v1.2.0.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+- Unable to select Harvester Cloud Provider in Rancher 2.7.x. See [#3750](https://github.com/harvester/harvester/issues/3750) [[Workaround]](https://github.com/harvester/harvester/issues/3750#issuecomment-1498757266).
+- Harvester upgrade may stuck in another operation (install/upgrade/rollback) by fleet. See [#3675](https://github.com/harvester/harvester/issues/3675) [[Workaround]](https://github.com/harvester/harvester/issues/3675#issuecomment-1533211282).
+- Upgrade issues: 
+    - Check https://docs.harvesterhci.io/dev/upgrade/v1-1-2-to-v1-2-0#known-issues for more information.
+    - An upgrade can't start and reports `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`. See [#3839](https://github.com/harvester/harvester/issues/3839) [[Workaround]](https://github.com/harvester/harvester/issues/3839#issuecomment-1534438192).
+    - An upgrade is stuck in `Creating Upgrade Repository`. See [#4246](https://github.com/harvester/harvester/issues/4246) [[Workaround]](https://github.com/harvester/harvester/issues/4246#issuecomment-1695107656)
```
Comment (LucasSaintarbor) [inline v1.2.0.md:113]:
```suggestion
    - An upgrade is stuck in `Creating Upgrade Repository`. See [#4246](https://github.com/harvester/harvester/issues/4246) [[Workaround]](https://github.com/harvester/harvester/issues/4246#issuecomment-1695107656).
```

**Review by LucasSaintarbor — APPROVED** [review]
Small NITs and updated `dev` URLs to `v1.2`

File: v1.2.0.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+- The system footprint of Harvester is reduced, better fitting Edge use cases. See [#3262](https://github.com/harvester/harvester/issues/3262).
+- Automatic adjustment of ResourceQuota during migration. When migrating VMs in Harvester, the resource limit of `ResourceQuota` is increased based on the specifications of the target VM before migration; and automatically decreased or restored to its previous value when the migration is complete. See [#3124](https://github.com/harvester/harvester/issues/3124).
+- Harvester supports adding an emulated Trusted Platform Module (TPM) to a VM. TPM is a cryptoprocessor that secures hardware using cryptographic keys and is a hard requirement of Windows 11. See [#2910](https://github.com/harvester/harvester/issues/2910).
+- Harvester supports passing through SR-IOV capable network virtual functions (VFs) to workload VMs. See [#2763](https://github.com/harvester/harvester/issues/2763).
+- Add `rancher-vcluster`(experimental) addon and bare-metal container support for the Harvester cluster. See [#2679](#https://github.com/harvester/harvester/issues/2679).
```
Comment (starbops) [inline v1.2.0.md:37]: Redundant `#` breaks the link.

```suggestion
- Add `rancher-vcluster`(experimental) addon and bare-metal container support for the Harvester cluster. See [#2679](https://github.com/harvester/harvester/issues/2679).
```

**Review by starbops — APPROVED** [review]
LGTM, just one nit

**Review by bk201 — APPROVED** [review]
LGTM, appreciate!

**Pull request merged**